pkg/api/ds_query_test.go

```diff
@@ -50,6 +52,66 @@ func (rv *fakePluginRequestValidator) Validate(dsURL string, req *http.Request)
 	return rv.err
 }
 
+// `/ds/query` endpoint test
+func TestAPIEndpoint_Metrics_QueryMetricsV2_ForwardHeaders(t *testing.T) {
+	forwardHeader1 := "Header1"
+	forwardHeader2 := "Header2"
+	dontForwardHeader := "Header3"
+	cfg := setting.NewCfg()
+	queryDataServiceWithAssert := func(assertQueryDataRequest func(req *backend.QueryDataRequest)) query.Service {
+		return query.ProvideService(
+			cfg,
+			nil,
+			nil,
+			&fakePluginRequestValidator{},
+			&fakePluginClient{
+				QueryDataHandlerFunc: func(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+					assertQueryDataRequest(req)
+					resp := backend.Responses{
+						"A": backend.DataResponse{
+							Status: http.StatusOK,
+						},
+					}
+					return &backend.QueryDataResponse{Responses: resp}, nil
+				},
+			},
+			plugincontext.ProvideService(cfg, localcache.ProvideService(), &pluginstore.FakePluginStore{
+				PluginList: []pluginstore.Plugin{
+					{
+						JSONData: plugins.JSONData{
+							ID: "grafana",
+						},
+					},
+				},
+			}, &fakeDatasources.FakeCacheService{}, &fakeDatasources.FakeDataSourceService{},
+				pluginSettings.ProvideService(dbtest.NewFakeDB(), secretstest.NewFakeSecretsService()), pluginconfig.NewFakePluginRequestConfigProvider()),
+		)
+	}
+
+	forwardHeadersAllowList = []string{forwardHeader1, forwardHeader2}
+	t.Run("Forward headers are added to the request to the plugin", func(t *testing.T) {
+		serverFeatureEnabled := SetupAPITestServer(t, func(hs *HTTPServer) {
+			assertQDR := func(req *backend.QueryDataRequest) {
+				assert.True(t, len(req.GetHTTPHeader(forwardHeader1)) > 0)
+				assert.True(t, len(req.GetHTTPHeader(forwardHeader2)) > 0)
+				assert.False(t, len(req.GetHTTPHeader(dontForwardHeader)) > 0)
```

Comment:
instead of asserting the header value is non-empty string, how about we assert the actual header values?